### DISCUSSION TITLE
[ADD] Prose checking

### DISCUSSION THREAD

**Discussion by lishaduck** [AndreaPontrandolfo/sheriff#338] [Ideas]
There's currently no linting for markdown, etc. This is a TS config, so I'd expect this to be out of scope, but I'm documenting it in case @AndreaPontrandolfo finds some of it useful.

- `npm:@eslint/markdown` (currently published as [`npm:eslint-plugin-markdown`](https://www.npmjs.com/package/eslint-plugin-markdown))
  - Adds linting to JS/TS code blocks in GFM Markdown.
  - Supports ESLint 9.
  - Is an official project, uses experimental ESLint language plugin support.
- [`npm:eslint-plugin-mdx`](https://www.npmjs.com/package/eslint-plugin-mdx)
  - Adds JS linting to MDX.
  - Adds JS/TS codeblock linting to MDX & GFM Markdown.
  - Supports Remark lints (including Prettier compatibility).
  - Does not officially support ESLint 9 (But works, except for GFM linting).
- [`npm:eslint-plugin-markdownlint`](https://www.npmjs.com/package/eslint-plugin-markdown)
  - Runs markdownlint, converts errors to ESLint format.
-  [`npm:@cspell/eslint-plugin`](https://www.npmjs.com/package/@cspell/eslint-plugin)
   - Spell checking for all programming languages, the ESLint plugin only checks strings, identifiers at declaration, and comments for JS/TS, preventing false positives over the CSpell CLI.

**Comment by AndreaPontrandolfo** [comment]
@lishaduck i converted this into a discussion because there are 4 different ideas/suggestions inside, and eventually every one of them should be treated as a different Issue.

**Comment by AndreaPontrandolfo** [comment]
About the [@cspell/eslint-plugin](https://www.npmjs.com/package/@cspell/eslint-plugin), the thing that worries me the most is the performance impact.

**Reply by lishaduck** [reply]
Yeah, I figure it could be optional/opt-in.
